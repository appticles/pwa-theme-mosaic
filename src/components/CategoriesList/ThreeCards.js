import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import CategoryCard from '../CategoryCard/index';

const ThreeCards = props => (
  <Grid columns={2} divided>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
  </Grid>
);

ThreeCards.propTypes = {
  categoriesList: PropTypes.array.isRequired,
};

export default ThreeCards;
