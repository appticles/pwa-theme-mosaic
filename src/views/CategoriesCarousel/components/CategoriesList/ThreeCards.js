import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';

const ThreeCards = props => (
  <Grid columns={2} divided className="half-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} key={props.categoriesList[1].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} key={props.categoriesList[2].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

ThreeCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default ThreeCards;
