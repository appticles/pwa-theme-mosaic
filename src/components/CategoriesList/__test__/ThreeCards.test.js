import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ThreeCards from '../ThreeCards';

describe('Three cards', () => {
  it('should render correctly', () => {
    const mockCategory = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
      {
        slug: 'mock-slug',
        id: 2,
        name: 'Some other mock name',
      },
      {
        slug: 'mock-slug',
        id: 3,
        name: 'Some name',
      },
    ];
    const output = shallow(<ThreeCards categoriesList={mockCategory} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
