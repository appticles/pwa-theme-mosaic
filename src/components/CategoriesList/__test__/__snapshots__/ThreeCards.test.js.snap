// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Three cards should render correctly 1`] = `
<Grid
  className="half-row"
  columns={2}
  divided={true}
>
  <GridRow>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 1,
            "name": "Some mock name",
            "slug": "mock-slug",
          }
        }
        key="1"
      />
    </GridColumn>
  </GridRow>
  <GridRow>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 2,
            "name": "Some other mock name",
            "slug": "mock-slug",
          }
        }
        key="2"
      />
    </GridColumn>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 3,
            "name": "Some name",
            "slug": "mock-slug",
          }
        }
        key="3"
      />
    </GridColumn>
  </GridRow>
</Grid>
`;
